new-log-viewer/src/components/DropFileContainer/index.tsx

```diff
@@ -0,0 +1,103 @@
+import React, {
+    useContext,
+    useState,
+} from "react";
+
+import {StateContext} from "../../contexts/StateContextProvider";
+import {CURSOR_CODE} from "../../typings/worker";
+
+import "./index.css";
+
+
+interface DropFileContextProviderProps {
+    children: React.ReactNode;
+}
+
+/**
+ * A container element to add drag & drop functionality to the child elements.
+ *
+ * @param props
+ * @param props.children
+ * @return
+ */
+const DropFileContainer = ({children}: DropFileContextProviderProps) => {
+    const {loadFile} = useContext(StateContext);
+    const [isFileHovering, setIsFileHovering] = useState(false);
+
+    /**
+     * Handler for a drag event.
+     *
+     * @param ev
+     */
+    const handleDrag = (ev: React.DragEvent<HTMLDivElement>) => {
+        ev.preventDefault();
+        ev.stopPropagation();
+
+        if ("dragenter" === ev.type) {
+            setIsFileHovering(true);
+        } else if ("dragleave" === ev.type) {
+            const {bottom, left, right, top} = ev.currentTarget.getBoundingClientRect();
+            if (ev.clientX >= left && ev.clientX <= right &&
+                ev.clientY >= top && ev.clientY <= bottom) {
+                return;
+            }
+
+            setIsFileHovering(false);
+        }
+    };
+
+    /**
+     * Handler for a drop event. handleFileDrop callback is used
+     * to load the dropped file into the viewer.
+     *
+     * @param ev
+     */
+    const handleDrop = (ev: React.DragEvent<HTMLDivElement>) => {
+        ev.preventDefault();
+        ev.stopPropagation();
+
+        setIsFileHovering(false);
+
+        const [file] = ev.dataTransfer.files;
+        if ("undefined" === typeof file) {
+            console.warn("No file is getting dropped");
+
+            return;
+        }
+        loadFile(file, {code: CURSOR_CODE.LAST_EVENT, args: null});
+    };
+
+    return (
+        <div
+            className={"drop-file-container"}
+            onDragEnter={handleDrag}
+            onDragLeave={handleDrag}
+            onDragOver={handleDrag}
+            onDrop={handleDrop}
+        >
+            <div
+                className={`drop-file-children ${isFileHovering ?
+                    "drop-file-children-no-pointer-events" :
```

Comment:
Is this obsolete or did we want to add this class?

Reply:
Right, it's no longer needed after we added the client range check on the dragleave event.